src/pages/the-elements-of-ui-engineering/index.pt-br.md

```diff
@@ -0,0 +1,61 @@
+
+---
+
+* **Consistência.** Você clica em um botão "Curtir" e o texto atualiza para: "Você e mais 3 amigos curtiram esse post." Você clica nele novamente e o texto retorna ao anterior. Parece fácil. Mas talvez um label assim está presente em diversos locais na tela. Talvez tenha uma outra indicação visual (como o fundo do botão) que deve ser alterado. A lista de "curtidores" que foi anteriormente obtida do servidor e está visível quando se passa por cima agora deve incluir seu nome. Se você navegar para uma outra página e voltar, o post não deve "esquecer" que foi curtido. Até mesmo a consistência local *por si só* cria uma lista de desafios. Mas outros usuários podem também modificar os dados que estamos mostrando (por exemplo ao curtir um post que estamos vendo). Como mantemos os mesmos dados sincronizados em diferentes partes da tela? Como e quando fazemos os dados locais serem consistentes com o servidor e como fazemos o caminho contrário?
+
+* **Responsividade.** As pessoas podem tolerar a falta de feedback visual de suas ações apenas por um tempo limitado. Para ações *contínuas* como gestos e rolagem da págnia, esse limite é bem baixo. (Até mesmo um pular um frame de 16ms faz parecer "desajeitado".) Para ações *discretas* como cliques, há pesquisas que dizem que os usuários acham que qualquer delay < 100ms é igualmente rápido. Mas há alguns desafios contra-intuitivos. Indicadores que fazem com que o layout "pule" ou que passe por diversos "estágios" de carregamento podem fazer com que a ação *pareça mais longa* do que foi. Da mesma forma, lidar uma interação em 20ms com o custo de descartar um frame de animação pode *parecer mais lento* que lidar com a mesma interação em 30ms sem descartas frames. Cérebros não são benchmarks. Como mantemos nossas aplicações responsivas para diferentes tipos de entradas? 
+
+* **Latência.** Ambos, processamento e acesso à rede gastam tempo. *Às vezes* podemos ignorar o custo computacional se não atrapalhar a responsividade em nossos dispositivos alvo (não se esqueça de testar sua aplicação em um dispositivo de baixa capacidade). Mas lidar com a latência da rede é inevitável - pode levar segundos! Nossa aplicação não pode apenas parar esperando os dados ou código serem carregados. Isso quer dizer que qualquer ação que depende de novos dados, códigos ou recursos é potencialmente assíncrona e precisa de lidar com o "carregamento". Mas isso pode acontecer em quase todas as telas. Como lidamos de forma elegante com a latência sem mostrar uma "cascata" de spinners ou "buracos" vazios? Como evitamos um layout "que pula"? E como trocamos as dependências assíncronas sem ter de "reconectar" nosso código toda vez?
```

Comment:
Just a suggestion. Maybe the use of "baixa performance" instead of "baixa capacidade" can have a better fit with "low-end devices".

```suggestion
* **Latência.** Ambos, processamento e acesso à rede gastam tempo. *Às vezes* podemos ignorar o custo computacional se não atrapalhar a responsividade em nossos dispositivos alvo (não se esqueça de testar sua aplicação em um dispositivo de baixa performance). Mas lidar com a latência da rede é inevitável - pode levar segundos! Nossa aplicação não pode apenas parar esperando os dados ou código serem carregados. Isso quer dizer que qualquer ação que depende de novos dados, códigos ou recursos é potencialmente assíncrona e precisa de lidar com o "carregamento". Mas isso pode acontecer em quase todas as telas. Como lidamos de forma elegante com a latência sem mostrar uma "cascata" de spinners ou "buracos" vazios? Como evitamos um layout "que pula"? E como trocamos as dependências assíncronas sem ter de "reconectar" nosso código toda vez?
```